tests/factories.py

```diff
@@ -128,6 +128,7 @@ def __init__(self):
             "prepare": set(),
             "execute": set(),
             "finalize": set(),
+            "finalize_log_only": set(),
```

Comment:
I am not convinced by the need for this separate tracker, it's awkward.

The trackers are meant to track if a api.FUNCTION is called or not, that is all.

Whether the job is cancelled or not seems irrelevent, just that it was called? We just testing that the run.py code calls the expected function. What that function does is out of for run.py tests I think?

Feels like this might a left over from earlier iteration, perhaps?